### PR TITLE
✨ [source-sentry] Allow low-code incremental streams to run within the Concurrent CDK framework

### PR DESCRIPTION
**Pull request by brianjlai**
## What

Adds support for the semi-incremental data feed streams to run in parallel with each other.

## How

Processing low-code incremental streams within the concurrent framework was introduced in https://github.com/airbytehq/airbyte/pull/46662 and version `6.0.0` of the CDK. Streams do not used windowed partitioning, but they can be run in parallel with each other.

Interpolation statements are also being updated to no longer use `stream_state` which might not be thread safe. We instead reference the `stream_interval` which is thread safe.

This PR upgrades the version to the latest to take advantage of concurrent processing. It also makes the necessary updates to `source.py` and `run.py` to pass incoming config/catalog/state when instantiating a source. This is a requirement of the concurrent CDK.

## User Impact

Technically none as all the backend processing changes shouldn't affect customers and we use default concurrency levels. It does however put some customers at a higher chance of encountering a rate limit.

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

We emit state in the legacy sequential format so that we can roll this back if needed. Once we start emitting concurrent state, it will be much harder to migrate backwards

### CODE REVIEW

File: airbyte-integrations/connectors/source-sentry/acceptance-test-config.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ acceptance_tests:
         skip_comprehensive_incremental_tests: true
         configured_catalog_path: integration_tests/configured_catalog.json
         future_state:
-          future_state_path: integration_tests/abnormal_state.json
+          bypass_reason: "Concurrent incremental streams emit a state with the most recent record or the lower boundary instead of the incoming abnormal state value"
```
Comment (brianjlai) [inline airbyte-integrations/connectors/source-sentry/acceptance-test-config.yml:35]: See slack convo for more context: https://airbytehq-team.slack.com/archives/C02U9R3AF37/p1730329381148769

basically now that we rely on the lower boundary of a date interval when checkpointing, we can no longer run the assertion on the abnormally large future state. This test is of dubious value so we may want to explore just tearing out the whole test. However, I don't want this to block the release of this connector

File: airbyte-integrations/connectors/source-sentry/unit_tests/integration/test_issues_stream.py [hunk 2]
```diff
@@ -31,24 +31,35 @@ def test_read(self, http_mocker: HttpMocker):
         http_mocker.get(
             HttpRequest(
                 url="https://sentry.io/api/0/projects/test%20organization/test%20project/issues/",
-                query_params={"query": "lastSeen:>1900-01-01T00:00:00.0Z"}
+                query_params={"query": "lastSeen:>1900-01-01T00:00:00.000000Z"}
```
Comment (brianjlai) [inline airbyte-integrations/connectors/source-sentry/unit_tests/integration/test_issues_stream.py:34]: i've tested this locally and sending in a micro second precision of 6 will work w/ sentry's API

**Comment by brianjlai** [comment]
regression test results:

I ran regression tests using two different scenarios:

- incremental sync of `issues` and `events` using state - https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/manual/brian_sentry_upgrade_to_concurrent_cdk/1730335086/a2bd0ff9ff64360b8f46f92f7a8cd9f6dd068af8/source-sentry/0.6.0/output.html
- `issues` and `events` without using state - https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/manual/brian_sentry_upgrade_to_concurrent_cdk/1730273699/29478ab7e383d83d3217f600b8b9300f808c9739/source-sentry/0.6.0/output.html

No mismatches. The other failing tests are expected because `is_resumable` is now included in the catalog. And for the field mismatches, we stopped include null fields in CDK 4, and since we're going from 0 -> 6, we expect this change

**Review by maxi297 — APPROVED** [review]
LGTM, thanks @brianjlai !

**Pull request merged**